packages/react-testing/src/root.tsx

```diff
@@ -95,19 +94,23 @@ export class Root<Props> implements Node<Props> {
     this.mount();
   }
 
-  act<T>(action: () => T, {update = true} = {}): T {
+  internalAct<T>(action: () => T, {update = true} = {}): T {
     const updateWrapper = update ? this.update.bind(this) : noop;
     let result!: T;
 
-    if (this.acting) {
-      return action();
-    }
+    reactAct(() => {
+      result = action();
+    });
+    updateWrapper();
+    return result;
+  }
 
-    this.acting = true;
+  act<T>(action: () => T, {update = true} = {}): T {
```

Comment:
? why both?

Reply:
I found that the issue I ran into was on all internal `act` would behave inconsistently and fail because of the way we were resolving promises.

Reply:
That said. The "external" `act` needs to resolve those promises for external async actions.

IF we didn't need to run `updateWrapper` I'd love to just use `react-dom/test-utils` directly.